polaris.shopify.com/content/patterns/date-picking.ts

```diff
@@ -435,387 +474,501 @@ const pattern: MultiVariantPattern = {
           return (
             <Popover
-            active={popoverActive}
-            autofocusTarget="none"
-            preferredAlignment="left"
-            preferredPosition="below"
-            fluidContent
-            fullHeight
-            activator={
-              <Button
-                icon={CalendarMinor}
-                onClick={() => setPopoverActive(!popoverActive)}
-              >
-                {buttonValue}
-              </Button>
-            }
-            onClose={() => setPopoverActive(false)}
-          >
-            <Box padding={{ xs: 2 }}>
-              <Columns columns={{ xs: "1fr", md: "1fr 4fr" }} ref={datePickerRef}>
-                <OptionList
-                  options={ranges.map((range) => ({
-                    value: range.alias,
-                    label: range.title,
-                  }))}
-                  selected={activeDateRange.alias}
-                  onChange={(value) => {
-                    setActiveDateRange(
-                      ranges.find((range) => range.alias === value[0])
-                    );
+              active={popoverActive}
+              autofocusTarget="none"
+              preferredAlignment="left"
+              preferredPosition="below"
+              fluidContent
+              sectioned={false}
+              fullHeight
+              activator={
+                <Button
+                  size="slim"
+                  icon={CalendarMinor}
+                  onClick={() => setPopoverActive(!popoverActive)}
+                >
+                  {buttonValue}
+                </Button>
+              }
+              onClose={() => setPopoverActive(false)}
+            >
+              {/* <Box padding={{ xs: 2 }} maxWidth={"728px"}> */}
+              <Popover.Pane fixed>
+                <Columns
+                  columns={{
+                    xs: "1fr",
+                    mdDown: "1fr",
+                    md: "max-content max-content",
                   }}
-                />
-                <Box padding={{ md: 6 }}>
-                  <AlphaStack fullWidth>
-                    <Inline>
-                      <div style={{ flexGrow: 1 }}>
-                        <TextField
-                          role="combobox"
-                          label={"Since"}
-                          labelHidden
-                          prefix={<Icon source={CalendarMinor} />}
-                          value={inputValues.since}
-                          onChange={handleStartInputValueChange}
-                          onBlur={handleInputBlur}
-                          autoComplete="off"
+                  gap={0}
+                  ref={datePickerRef}
+                >
+                  <Box
+                    maxWidth={mdDown ? "516px" : "212px"}
+                    width={mdDown ? "100%" : "212px"}
+                    padding={{ xs: 5, md: 0 }}
+                    paddingBlockEnd={{ xs: 1, md: 0 }}
+                  >
```

Comment:
Oooft, this is a lot of detail that feels very fragile, and opens up a whole bunch of questions:

- Where does `212px` come from?
- What happens if a builder modifies one of these values?
- Are these values _opinions_ of the pattern, or just demo values so it looks pretty?

Reply:
This is a magic pixels currently embedded in sass in the reference implementation.
I'm happy that Box lets us **not** write this in sass, but for sure the magic pixel is not ideal. 

We **could** try and make this more explicit as preview prettification by putting this up into a commented variable, but imo wouldn't solve the root problem and would probably make the code even more confusing.